src/jimgw/samplers/blackjax/smc.py

```
@@ -156,65 +162,91 @@ def mcmc_parameter_update_fn(_key, state, _info):
-        def cond_fn(carry: tuple) -> Any:
-            s = carry[0]
-            return s.sampler_state.tempering_param < 1.0
+        accept_h = np.zeros(max_iterations)
+        cov_scale_h = np.zeros(max_iterations)
+        cov_scale = float(config.initial_cov_scale)
+        n_iter = 0
+
+        # Resume from checkpoint if one exists.
+        if ckpt_path is not None and ckpt_path.exists():
+            with open(ckpt_path, "rb") as _f:
+                _ckpt = pickle.load(_f)
```

Ruff (0.15.14)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)